src/datetime/mod.rs

```diff
@@ -79,11 +79,141 @@ pub enum SecondsFormat {
+/// information needed to exactly specify a point in time, and to do calculations on it: a date,
+/// time, offset from UTC, and timezone.
 ///
-/// There are some constructors implemented here (the `from_*` methods), but
-/// the general-purpose constructors are all via the methods on the
-/// [`TimeZone`](./offset/trait.TimeZone.html) implementations.
+/// Internally it stores the information as a UTC [`NaiveDateTime`] and a generic type implementing
+/// [`Offset`]. The offset contains a type with the offset from UTC, usually as a value. The type
+/// implementing the [`TimeZone`] is tracked by the type system.
+///
+/// In the case of `DateTime<Utc>` the offset is a zero-sized type. This makes a `DateTime<Utc>`
+/// the same size as a [`NaiveDateTime`], while encoding in the type system this value is in UTC
+/// instead of a localized time.
+///
+/// # Timezones
+///
+/// [`DateTime`] is generic over types implementing [`TimeZone`].
+///
+/// [`TimeZone`] implementations differ in the amount of information they have available, and
+/// because of that the range of operations they can do correctly.
+///
+/// Ordered from most to least information:
+/// - Types implementing proper timezone support using the IANA time zone database, such as in the
+///   [`chrono_tz`](https://crates.io/crates/chrono_tz`) and [`tzfile`](
+///   https://crates.io/crates/tzfile) crates. They support changes like offset shifts and daylight
+///   saving time.
+/// - [`Local`] matches the current timezone settings of the operating system. That may however be a
+///   simplified implementation, not describing all complexities like the timezone database does.
+/// - [`FixedOffset`] stores only the offset from UTC of a specific datetime, without knowing the
+///   timezone. This is the preferred type for an end result, a timestamp.
+/// - [`Utc`] is the universal time, carrying no information about a local time at all. It is
+///   continuous, without gaps or ambiguities.
+///
+/// # Creating a `DateTime`
+///
+/// ## Combine from primitive types
+/// Methods are available on multiple types for convenience.
+/// - Combine a [`TimeZone`] and a _local_ [`NaiveDateTime`]:
+///   - [`NaiveDateTime::and_local_timezone`]
+///   - [`TimeZone::from_local_datetime`]
+///   - [`DateTime<Tz: TimeZone>::from_local`]
+/// - Combine a [`TimeZone`] and a _UTC_ [`NaiveDateTime`]:
+///   - [`TimeZone::from_utc_datetime`]
+///   - [`DateTime<Tz: TimeZone>::from_utc`]
+/// - (convenience) Combine a [`TimeZone`] with _local_ year, month, day and time components:
+///   - [`TimeZone::with_ymd_and_hms`]
+///
+/// ## From the current time
+/// (Requires the default `clock` feature.)
+/// - [`Local.now()`](offset/struct.Local.html#method.now)
+/// - [`Utc.now()`](offset/struct.Utc.html#method.now)
+///
+/// ## Parsing
+/// - [`parse_from_str`](#method.parse_from_str) for parsing with a format string.
+/// - [`parse_from_rfc2822`](#method.parse_from_rfc2822) for parsing a string such as
```

Comment:
A precise link to the official RFC will lend trustworthiness to these docs.

```rust
/// ... for parsing an [RFC 2822] compliant string such as ...
///
/// [RFC 2822]: https://www.rfc-editor.org/rfc/rfc2822.html#section-3.3
```

Reply:
Excellent idea.